apollo-router-core/src/spec/field_type.rs

```diff
@@ -0,0 +1,119 @@
+use crate::*;
+use apollo_parser::ast;
+
+#[derive(Debug)]
+pub(crate) struct InvalidValue;
+
+// Primitives are taken from scalars: https://spec.graphql.org/draft/#sec-Scalars
+#[derive(Debug)]
+pub(crate) enum FieldType {
+    Named(String),
+    List(Box<FieldType>),
+    NonNull(Box<FieldType>),
+    String,
+    Int,
+    Float,
+    Id,
+    Boolean,
+}
+
+impl FieldType {
+    pub(crate) fn validate_value(
```

Comment:
Float, Integer and Bool can be represented as strings, so coercion supported need to be added. Can be a followup ticket though.

Reply:
what, where did you see that? oh that would explain what i saw with the ID type

Reply:
Fixed in 41e2279c82635167523d7aa2768caf40ff69fcc2